tests/app/dao/test_templates_dao.py

```diff
@@ -957,3 +959,67 @@ def test_template_with_provider_id_persists_provider_id(
     assert notify_db_session.session.get(Template, template.id).provider_id == provider.id
     # Teardown
     template_cleanup(notify_db_session.session, template.id)
+
+
+class TestDAOGetTemplateById:
+    def test_dao_get_template_by_id_with_no_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        result = dao_get_template_by_id(template.id)
+
+        assert result == template
+        assert isinstance(result, Template)
+        assert result.id == template.id
+        assert result.version == 1
+
+    def test_dao_get_template_by_id_with_specific_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        template.content = 'Updated content'
+        dao_update_template(template)
+
+        result = dao_get_template_by_id(template.id, version=1)
+
+        assert isinstance(result, TemplateHistoryData)
+        assert result.id == template.id
+        assert result.version == 1
+        assert result.content != 'Updated content'
```

Comment:
But does it equal the pre-update value?  I would store that value before the update and use it here.

Reply:
I'm suggesting something like this:
```python
template = sample_template()
template_content = template.content
...
assert result.content == template_content
```
I also suggest turning what you wrote in the above comment into a comment in the code.  It isn't obvious to a new team member that `dao_update_template` creates a new version.